services/elasticache/pkg/resource/cache_parameter_group/custom_api.go

```diff
@@ -52,9 +53,58 @@ func (rm *resourceManager) customSetOutputDescribeCacheParameters(
 		return err
 	}
 	ko.Status.Parameters = parameters
+	rm.customSetOutputSupplementAPIs(ctx, cacheParameterGroupName, ko)
+
 	return nil
 }
 
+func (rm *resourceManager) customSetOutputSupplementAPIs(
+	ctx context.Context,
+	cacheParameterGroupName *string,
+	ko *svcapitypes.CacheParameterGroup,
+) (bool, error) {
+	events, err := rm.provideEvents(ctx, cacheParameterGroupName, 20)
+	if err != nil {
+		return false, err
+	}
+	ko.Status.Events = events
+	return true, nil
+}
+
+func (rm *resourceManager) provideEvents(
+	ctx context.Context,
+	cacheParameterGroupName *string,
+	maxRecords int64,
+) ([]*svcapitypes.Event, error) {
+	input := &svcsdk.DescribeEventsInput{}
+	input.SetSourceType("cache-parameter-group")
+	input.SetSourceIdentifier(*cacheParameterGroupName)
+	input.SetMaxRecords(maxRecords)
+	input.SetDuration(20160) // max 14 days in minutes
```

Comment:
nit: consider making this a constant instead of hard-coding 20160

Reply:
updated the code and declared a constant.